In [ ]:
#Q1
First_Name = "Jaya krishna"
last_name = 'Reddy'
Name = First_Name + ' ' + last_name
parent_name = 'Ramana Reddy '
Age = 23
Acadamics = "B-tech Ece"
Address = 'Road_Num 8, Tkr_kaman,Srishakthi_nagar,500079'
Birth_Date = '22/sep/2001'

print(f'Name : {Name}')
print(f'parent_name : {parent_name}')
print(f'Age : {Age}')
print(f'Acadamics : {Acadamics}')
print(f'Address : {Address}')
print(f'Birth_Date : {Birth_Date}')

Name : Jaya krishna Reddy
parent_name : Ramana Reddy 
Age : 23
Acadamics : B-tech Ece
Address : Road_Num 8, Tkr_kaman,Srishakthi_nagar,500079
Birth_Date : 22/sep/2001


In [ ]:
#Q2
surname = "policepatel"
print(f'{surname} {Name}')

policepatel Jaya krishna Reddy


#### Q3


*  Variables are selected reasonably so that someone reviewing or testing the code could understand

*   No spaces should be used while nameing  the variables
*   Variable are not encourage to start with numbers


*   One should not select variable names from predefined functions or keywords selecting them could result in error in the function of predefined functions



In [ ]:
#Q3b
a = 4.5
b = "Innomatics"
c = 100
d = True
print (type(a),type(b),type(c),type(d))

<class 'float'> <class 'str'> <class 'int'> <class 'bool'>


In [ ]:
#Q4
Break = 16
print(Break)

16


Break is a Predefined keyword which is used to break a loop so using this as a variable name could lead error while running a loop with break function

In [ ]:
#Q5
Length = 52
width = 88.7
Area = Length * width
print(f'{Area} mr**2')


4612.400000000001 mr**2


#Operators

In [ ]:
#Q1
a,b =12,5
print(f'sum of {a} , {b} is {a+b}')
print(f'difference of {a} , {b} is {a-b}')
print(f'product of {a} , {b} is {a*b}')
print(f'quotient of {a} , {b} is {a//b}')
print(f'remainder of {a} , {b} is {a%b}')



sum of 12 , 5 is 17
difference of 12 , 5 is 7
product of 12 , 5 is 60
quotient of 12 , 5 is 2
remainder of 12 , 5 is 2


In [ ]:
#Q2
p = True
q = False
r = True

print(f"p and q: {p and q}")
print(f"p or q: {p or q}")
print(f"not r: {not r}")

p and q: False
p or q: True
not r: False


In [ ]:
#Q3
Expression = 10 + 2 * 3 / 4 - 5 ** 2
print(Expression)

-13.5


### Order of Execution

*   uses BODMAS OR PEMDAS
In BODMAS the priority while solving a problem starts with brackets,order,division,multiplication,addtion and subtraction  
In PEMDAS it starts with Parentheses, Exponents, Multiplication, Division, Addition, and Subtraction.

so according to this the solution process is


1.  Exponentiation (**) : 5 ** 2 = 25

1.   Multiplication (*) : 2 * 3 = 6
2.   Division (/) : 6 / 4 = 1.5
2. Addition (+) : 10 + 1.5 = 11.5

1.   subtraction (-) : 11.5 - 25 = -13.5


In [ ]:
#4
expression = (8 + 2) * (3 - 1) ** 2 / 2
print(expression)


20.0


1.   As explained above the Process of PEMDAS is followed

1.   Parentheses override the default operator process and ensures that the problem is solved from inside then continues with other
3.   If parentheses are not used then the problem would first start with power values then the solution would be totally different




In [ ]:
#Q5
a = 5
b = 5
print(f"a is b: {a is b}")
print(f"a is not b: {a is not b}")

a is b: True
a is not b: False


In [ ]:
#Q6
a = True
b = False
c = True

x = (a and b) or (not c)
print(x)

False


In [4]:
#7
x = input("Enter any string : ")
lx =len(x)
if lx%2==0:
  print(x[(lx//2)-1:(lx//2)+1])
else:
  print(x[::lx//2])

Enter any string : holiday
hiy


In [1]:
#7b
x = input('enter a string of even len:')
lx =len(x)
x[(lx//2)-1:(lx//2+1)]

enter a string of even len:innomatics


'ma'

# Conditional statements

In [ ]:
#Q1
Number = input("Enter any desired number : ")

last_digit = int(Number[-1])
if last_digit % 3 == 0:
  print(f' last digit of {Number} is divisible by 3')
else:
  print(f'last digit of {Number} is not divisible by 3')


Enter any desired number : 23
 last digit of 23 is divisible by 3


In [ ]:
#Q2
Salary =int(input("Enter your salary : "))
Experience = int(input("Enter your Experience : "))

if Experience >= 5:
   Revised_sal = Salary *(1+ 5/100)
   print(f'Revised salary is {Revised_sal}')
else:
  print(f'wait for {5-Experience} year/s')

Enter your salary : 120000
Enter your Experience : 6
Revised salary is 126000.0


In [ ]:
#Q3
Score = int(input("Enter your marks : "))
if Score >80 :
  print("Grade A")
elif 60 < Score <= 80:
  print("Grade B")
elif 50 < Score <= 60:
  print("Grade C")
elif 45 < Score <= 50:
  print("Grade D")
elif 25 < Score <= 45:
  print("Grade E")
else:
  print("Grade F")

Enter your marks : 77
Grade B


In [ ]:
#Q4
A = int(input("Enter 1st number : "))
B = int(input("Enter 2nd number : "))
C = int(input("Enter 3rd number : "))
average = (A+B+C)/3
print(average)

Enter 1st number : 2
Enter 2nd number : 3
Enter 3rd number : 4
3.0


In [ ]:
#Q5
A = int(input("Enter 1st number : "))
B = int(input("Enter 2nd number : "))
if A>B:
  print(f'{A+B}')
else:
  print(f'{B-A}')

Enter 1st number : 12
Enter 2nd number : 32
20


In [ ]:
#6
Number = int(input("Enter any number : "))
if Number%3 ==0 and Number%5 ==0:
  print(f'{Number} is divisible by both 3 and 5')
else:
  print(f'{Number} is not divisible by both 3 and 5')

Enter any number : 15
15 is divisible by both 3 and 5


In [ ]:
#Q7
Number = int(input("Enter any number : "))
if 10<= Number >= 50 and not (20<= Number >= 30):
  print(f'{Number} is in range')
else:
  print(f'{Number} is not in range')

Enter any number : 11
11 is not in range


In [ ]:
#Q8
Number = int(input("Enter any number : "))
if Number > 10**2 :
  print(f'{Number} is grater than the square of 10 ')
elif Number < 10**2 :
  print(f'{Number} is less than the square of 10 ')
else:
  print(f'{Number} is equal to the square of 10 ')


Enter any number : 121
121 is grater than the square of 10 


In [ ]:
#9
Person_1 =int(input("Enter your age : "))
Person_2 = int(input("Enter your age : "))
if Person_1 > Person_2:
  print(f'Person_1 is elder than Person_2')
elif Person_1 < Person_2:
  print(f'Person_2 is elder than Person_1')
else:
  print("same Age!")

Enter your age : 12
Enter your age : 12
same Age!


In [ ]:
#10
A = int(input("Enter 1st number : "))
B = int(input("Enter 2nd number : "))
C = int(input("Enter 3rd number : "))
list_of_ages = [A,B,C]
sort = sorted(list_of_ages,reverse =True)
print(sort)

Enter 1st number : 2
Enter 2nd number : 3
Enter 3rd number : 1
[3, 2, 1]


In [ ]:
#11
units = int(input("Enter the units consumed : "))
if units<= 100:
  bill = 100 * 5
  print(f'Total bill is {bill}')
elif 101 <= units <= 300:
  bill = 100 * 5 + (units - 100) * 7
  print(f'Total bill is {bill}')
elif units>300:
  bill = 100 * 5 + 200 * 7 + (units - 300) * 10
  print(f'Total bill is {bill}')



Enter the units consumed : 323
Total bill is 2130


In [ ]:
#12
A = int(input('Enter length of first side : '))
B = int(input('Enter length of second side : '))
C = int(input('Enter length of third side :'))

if A+B <C or A+C < B or B+C < A:
  print('Triangle is not possible')
else:
  print('Triangle is possible')

Enter length of first side : 12
Enter length of second side : 2
Enter length of third side :3
Triangle is not possible


In [ ]:
#13
A = int(input('Enter length of first side : '))
B = int(input('Enter length of second side : '))
C = int(input('Enter length of third side :'))
if A==B==C:
  print("It's an Equilateral Triangle All sides are equal")
elif A==B or A==C or B==C:
  print("It's an Isosceles Triangle only two sides are equal")
else:
  print(" It's a Scalene Triangle no sides are equal")

Enter length of first side : 2
Enter length of second side : 2
Enter length of third side :2
It's an Equilateral Triangle All sides are equal


In [ ]:
lx=0
assert lx!=0, 'please provide a string of len grater than o'